src/ceph-volume/ceph_volume/devices/lvm/zap.py

```diff
@@ -41,6 +42,79 @@ def zap_data(path):
     ])
 
 
+def find_associated_devices(osd_id=None, osd_fsid=None):
+    """
+    From an ``osd_id`` and/or an ``osd_fsid``, filter out all the LVs in the
+    system that match those tag values, further detect if any partitions are
+    part of the OSD, and then return the set of LVs and partitions (if any).
+    """
+    lv_tags = {}
+    if osd_id:
+        lv_tags['ceph.osd_id'] = osd_id
+    if osd_fsid:
+        lv_tags['ceph.osd_fsid'] = osd_fsid
+    lvs = api.Volumes()
+    lvs.filter(lv_tags=lv_tags)
+    if not lvs:
```

Comment:
How do we handle the case where the osd fsid matches but not the osd id and the other way around?

Reply:
They both have to match. If an admin is passing both id and fsid, it means that both need to match.